### PR TITLE
Cache genesis setup in App.Setup

### PR DESCRIPTION
**Pull request by ValarDragon**
## What is the purpose of the change

cref #2148 , small speedup to app setup time / all tests

```
BEFORE
BenchmarkSetup-16             57          20107194 ns/op         1753442 B/op      22513 allocs/op

AFTER
BenchmarkSetup-16    	      76	  16989387 ns/op	 1087661 B/op	   13536 allocs/op
```

## Brief Changelog

deduplicate genesis creation & encoding config every time its needed

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

File: app/apptesting/test_suite.go [hunk 1]
```diff
@@ -71,7 +71,9 @@ func (s *KeeperTestHelper) SetEpochStartTime() {
 		epoch.StartTime = s.Ctx.BlockTime()
 		epochsKeeper.DeleteEpochInfo(s.Ctx, epoch.Identifier)
 		err := epochsKeeper.AddEpochInfo(s.Ctx, epoch)
-		s.Require().NoError(err)
+		if err != nil {
+			panic(err)
+		}
```
Comment (p0mvn) [inline app/apptesting/test_suite.go:76]: Why this change?

**Review by p0mvn — APPROVED** [review]

**Review by czarcas7ic — APPROVED** [review]
Simple fix, nice 😄

File: app/apptesting/test_suite.go (hunk 1 above)
Comment (ValarDragon) [reply app/apptesting/test_suite.go:76]: We can't use any suite methods for the benchmark (until #2554 )
